engine/src/main/java/org/hibernate/search/engine/backend/work/execution/spi/IndexWorkspace.java

```diff
@@ -14,12 +14,32 @@
  */
 public interface IndexWorkspace {
 
-	CompletableFuture<?> mergeSegments();
+	@Deprecated
+	default CompletableFuture<?> mergeSegments() {
+		return mergeSegments( OperationSubmitter.DEFAULT );
+	}
 
-	CompletableFuture<?> purge(Set<String> routingKeys);
+	CompletableFuture<?> mergeSegments(OperationSubmitter operationSubmitter);
```

Comment:
I'm a bit surprised you decided to add this parameter to `IndexWorkspace` in the end, since during our last discussion we agreed it was not possible to implement support for `operationSubmitter` in this interface for Lucene. But ok, let's see where this leads us...